### ISSUE TITLE
DTO between service and repository

### ISSUE THREAD

**Issue by n-r-w** [ruslantsyganok/clean_arcitecture_golang_example#2]
Shouldn't we convert datastructure (aka entity) into another kind of DTO at the transition between service (aka usecase) and repository(aka gateway)? After all, the rules of interaction between business logic and the database can change as well as the rules at the handler level.

**Comment by ruslantsyganok** [comment]
Yes, of course we do! This repo (or video for that matter) is fairly old!
Thanks for your comment!

**Comment by Pulgafree** [comment]
I will like to see and follow up video, updating this matter and going from zero to complete example.

Much like learn by doing approach
